### PR TITLE
Intelligently complete return with `;` or ` `

### PR DESCRIPTION
**Pull request by matklad**
r? @alygin

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/completion/RsKeywordCompletionProvider.kt [hunk 1]
```diff
@@ -36,3 +27,20 @@ class AddSuffixInsertionHandler(val suffix: String) : InsertHandler<LookupElemen
...
+private val ALWAYS_NEEDS_SPACE = setOf("crate", "const", "enum", "extern", "fn", "impl", "let", "mod", "mut", "pub",
+    "static", "struct", "trait", "type", "unsafe", "use")
+
+
+private fun addInsertionHandler(keyword: String, builder: LookupElementBuilder, parameters: CompletionParameters): LookupElementBuilder {
+    val suffix = when (keyword) {
+        in ALWAYS_NEEDS_SPACE -> " "
+        "return" -> {
+            val fn = parameters.position.parentOfType<RsFunction>() ?: return builder
+            if (fn.retType != null) " " else ";"
```
Comment (alygin) [inline src/main/kotlin/org/rust/lang/core/completion/RsKeywordCompletionProvider.kt]: Inside `fn foo() -> ()` it could also be completed with `;`

Diverging functions probably also need a special case (don't suggest `return` at all), but that may be a subject for a separate PR.

Comment (matklad) [inline src/main/kotlin/org/rust/lang/core/completion/RsKeywordCompletionProvider.kt]: Good idea!

**Comment by matklad** [comment]
bors r=alygin
